crates/wasm-smith/src/code_builder.rs

```diff
@@ -508,18 +508,23 @@ where
         let mut instructions = vec![];
 
         while !self.allocs.controls.is_empty() {
-            let keep_going =
-                instructions.len() < max_instructions && u.arbitrary().unwrap_or(false);
+            let keep_going = instructions.len() < max_instructions
+                && u.arbitrary().map_or(false, |b: u8| b != 0);
             if !keep_going {
                 self.end_active_control_frames(&mut instructions);
                 break;
             }
 
             gather_options(module, &mut self);
-
-            let f = u.choose(&self.allocs.options)?;
-            let inst = f(u, module, &mut self)?;
-            instructions.push(inst);
+            if let Ok(f) = u.choose(&self.allocs.options) {
+                // `u.choose` can fail for two reasons and in both we swallow the error:
+                // - it fails when no options were provided, so we continue looping until we run out
+                //   of random data,
+                // - or it fails because there is not enough underlying data, so we really cannot
+                //   generate an instruction.
```

Comment:
Note that the first is impossible in this particular case, because we can always add an `unreachable` or a `nop` instruction, for example.

Reply:
(And I think it is worth clarifying that in this comment)

Reply:
Ok, perhaps I can add the comment and `assert` that there are always options available?

Reply:
Perfect :)